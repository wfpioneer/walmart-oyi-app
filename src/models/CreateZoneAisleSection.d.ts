import { CreateAisles } from '../state/reducers/Location';

export interface CreateAisleRequest {
  zoneId: number;
  aisles: CreateAisles[]
}

export interface ZoneAislesSectionResponse {
  aisleId: number;
  aisleName: number;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
  sectionCount: number;
  status: number;
}

export interface CreateZoneAisleSectionResponse {
  aisles: ZoneAislesSectionResponse[];
  status: number;
  zoneId: number;
  zoneName: string;
}

export interface CreateAisleResponse {
  aisleId: number;
  aisleName: number;
  status: number;
  message: string;
  createdSectionsResponseDto: CreateAisleSectionsResponse;
}

interface CreateAisleSectionsResponse {
  aisleId: number;
  status: number;
  message: string;
  createdSections: string[]
}
