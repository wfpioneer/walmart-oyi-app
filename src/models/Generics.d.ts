import { EventArg } from '@react-navigation/native';
import { EffectCallback } from 'react';

export type UseStateType<T> = [T, React.Dispatch<React.SetStateAction<T>>];
export type UseEffectType = (effect: EffectCallback, deps?: ReadonlyArray<any>) => void;

/**
 * Helper type for bringing track event information from the parent screen
 * into components where the component accepts user interaction for the parent screen
 */
export type TrackEventSource = {
  screen: string;
  action: string;
  otherInfo?: any;
};

export type BeforeRemoveEvent = EventArg<'beforeRemove', true, {
  action: Readonly<{
      type: string;
      payload?: object | undefined;
      source?: string | undefined;
      target?: string | undefined;
  }>;
}>;

export type ScannedEvent = {
  value: any;
  type: string | null;
};
