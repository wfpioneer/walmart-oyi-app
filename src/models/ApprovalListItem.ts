export interface ApprovalListItem {
  imageUrl?: string;
  itemName: string;
  itemNbr: number;
  upcNbr: number;
  categoryNbr: number;
  categoryDescription: string;
  subCategoryNbr: number;
  subCategoryDescription: string;
  oldQuantity: number;
  newQuantity: number;
  dollarChange: number;
  initiatedUserId: string;
  initiatedTimestamp: string;
  approvalStatus: approvalStatus;
  approvalRequestSource: approvalRequestSource;
  isChecked?: boolean;
  daysLeft?: number;
}

export interface ApprovalCategory extends ApprovalListItem {
  categoryHeader?: boolean;
}

export enum approvalStatus {
  'Pending' = 'Pending',
  'Approved' = 'Approved',
  'Rejected' = 'Rejected',
  'Expired' = 'Expired'
}
export enum approvalRequestSource {
  'ItemDetails' = 'itemdetails',
  'Audits' = 'audits',

}
