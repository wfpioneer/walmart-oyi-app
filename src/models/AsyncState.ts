export interface AsyncState{
  isWaiting: boolean;
  value: any;
  error: any;
  result: any;
}
