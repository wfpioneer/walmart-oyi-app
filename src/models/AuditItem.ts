export interface ItemPalletInfo {
  palletId: string,
  quantity: number,
  sectionId: number,
  locationName: string,
  mixedPallet: boolean,
  newQty?: number
}
