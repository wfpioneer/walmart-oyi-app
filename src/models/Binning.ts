export interface BinningItem {
  itemNbr: number;
  itemDesc: string;
  price: number;
  upcNbr: string;
  quantity: number;
}
export interface BinningPallet {
  id: number;
  expirationDate: string;
  lastLocation?: string;
  firstItem: BinningItem;
}