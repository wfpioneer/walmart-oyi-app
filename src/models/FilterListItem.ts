interface FilterListItem {
    value: string;
    display: string;
    selected: boolean;
  }

export default FilterListItem;
