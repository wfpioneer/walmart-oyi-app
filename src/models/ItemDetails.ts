import Location from './Location';
import { approvalRequestSource, approvalStatus } from './ApprovalListItem';

export interface IOHChangeHistory {
  id : number;
  itemName : string;
  itemNbr : number;
  upcNbr : number;
  categoryNbr : number;
  categoryDescription : string;
  subCategoryNbr : number;
  subCategoryDescription : string;
  oldQuantity : number;
  newQuantity : number;
  dollarChange : number;
  daysLeft: number;
  initiatedUserId : string;
  initiatedTimestamp : string;
  approvalStatus : approvalStatus;
  approvalRequestSource : approvalRequestSource;
}

export interface IPickHistory {
  id:number,
  itemNbr: number,
  upcNbr: number,
  itemDesc: string,
  itemQty: number,
  category: string,
  quickPick: boolean,
  salesFloorLocationName: string,
  salesFloorLocationId: number,
  moveToFront: boolean,
  assignedAssociate: string,
  palletId: number,
  palletLocationName: string,
  palletLocationId: number,
  status: string,
  createdBy: string,
  createTS: string
}

interface ItemDetails {
  itemName: string;
  itemNbr: number;
  upcNbr: string;
  status?: string;
  categoryNbr: number;
  categoryDesc: string;
  price: number;
  basePrice: number
  exceptionType?: string; // This is enumerated
  imageUrlKey?: string;
  imageBlobKey?: string;
  completed: boolean;
  onHandsQty: number;
  pendingOnHandsQty: number;
  consolidatedOnHandQty: number;
  claimsOnHandQty: number;
  backroomQty: number;
  cloudQty?: number;
  inTransitCloudQty?: number;
  replenishment: {
    onOrder: number;
  };
  location: {
    floor?: Location[];
    reserve?: Location[];
    count: number;
  };
  sales: {
    lastUpdateTs: string;
    dailyAvgSales: number;
    daily: {
      day: string;
      value: number;
    }[];
    weeklyAvgSales: number;
    weekly: {
      week: number;
      value: number;
    }[];
  };
  ohChangeHistory?: IOHChangeHistory[];
  picklistHistory?: IPickHistory[]
}

export default ItemDetails;
