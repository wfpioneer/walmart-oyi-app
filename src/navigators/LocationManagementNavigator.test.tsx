import React from 'react';
import { NavigationProp } from '@react-navigation/native';
import ShallowRenderer from 'react-test-renderer/shallow';
import {
  LocationManagementNavigatorStack, renderCamButton, renderScanButton, resetLocManualScan
} from './LocationManagementNavigator';
import { AsyncState } from '../models/AsyncState';

jest.mock('../utils/AppCenterTool.ts', () => jest.requireActual('../utils/__mocks__/AppCenterTool'));
jest.mock('../utils/sessionTimeout.ts', () => jest.requireActual('../utils/__mocks__/sessTimeout'));
let navigationProp: NavigationProp<any>;

describe('LocationManagement Navigator', () => {
  const defaultAsyncState: AsyncState = {
    isWaiting: false,
    error: null,
    value: null,
    result: null
  };
  // TODO we are not able to currentlt test the navigator's headers through simple snapshot tests
  it('Renders the LocationManagement Navigator, non manager', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationManagementNavigatorStack
        isManualScanEnabled={false}
        dispatch={jest.fn()}
        navigation={navigationProp}
        userFeatures={['location management edit']}
        locationPopupVisible={false}
        getSectionDetailsApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the LocationManagement Navigator, manager', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationManagementNavigatorStack
        isManualScanEnabled={false}
        dispatch={jest.fn()}
        navigation={navigationProp}
        userFeatures={['manager approval', 'location management edit']}
        locationPopupVisible={false}
        getSectionDetailsApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the LocationManagement Navigator with LocationKebabMenu disabled', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationManagementNavigatorStack
        isManualScanEnabled={false}
        dispatch={jest.fn()}
        navigation={navigationProp}
        userFeatures={[]}
        locationPopupVisible={false}
        getSectionDetailsApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the LocationManagement Navigator with Print List Button', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <LocationManagementNavigatorStack
        isManualScanEnabled={false}
        dispatch={jest.fn()}
        navigation={navigationProp}
        userFeatures={['location printing']}
        locationPopupVisible={false}
        getSectionDetailsApi={defaultAsyncState}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the cameraButton header icon', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      renderCamButton()
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Renders the scanButton header icon', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      renderScanButton(jest.fn(), false)
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('Expects dispatch to be called if isManualScanEnabled is true for "resetLocManualScan()"', () => {
    const mockDispatch = jest.fn();
    const manualScanEnabled = true;
    resetLocManualScan(manualScanEnabled, mockDispatch);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
