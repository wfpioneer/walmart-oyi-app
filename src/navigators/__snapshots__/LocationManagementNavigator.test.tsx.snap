// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add card render tests renders the bottom sheet add card, not visible 1`] = `null`;

exports[`Add card render tests renders the bottom sheet add card, visible, manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;

exports[`Add card render tests renders the bottom sheet add card, visible, non manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;

exports[`LocationManagement Navigator Renders the LocationManagement Navigator, manager 1`] = `
<StackNavigator
  headerMode="float"
  screenOptions={
    Object {
      "headerStyle": Object {
        "backgroundColor": "#0071e9",
      },
      "headerTintColor": "#ffffffff",
    }
  }
>
  <Screen
    component={[Function]}
    name="Zones"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.LOCATION.ZONES\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Aisles"
    options={
      Object {
        "headerTitle": "[missing \\"en.LOCATION.AISLES\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Sections"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.LOCATION.SECTIONS\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="LocationDetails"
    options={
      Object {
        "headerTitle": "[missing \\"en.LOCATION.LOCATION_DETAILS\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="AddLocation"
    options={
      Object {
        "headerBackTitleVisible": false,
        "headerTitle": "[missing \\"en.LOCATION.ADD_NEW_LOCATION\\" translation]",
        "headerTitleAlign": "left",
        "headerTitleStyle": Object {
          "fontSize": 18,
        },
      }
    }
  />
</StackNavigator>
`;

exports[`LocationManagement Navigator Renders the LocationManagement Navigator, non manager 1`] = `
<StackNavigator
  headerMode="float"
  screenOptions={
    Object {
      "headerStyle": Object {
        "backgroundColor": "#0071e9",
      },
      "headerTintColor": "#ffffffff",
    }
  }
>
  <Screen
    component={[Function]}
    name="Zones"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.LOCATION.ZONES\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Aisles"
    options={
      Object {
        "headerTitle": "[missing \\"en.LOCATION.AISLES\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="Sections"
    options={
      Object {
        "headerRight": [Function],
        "headerTitle": "[missing \\"en.LOCATION.SECTIONS\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="LocationDetails"
    options={
      Object {
        "headerTitle": "[missing \\"en.LOCATION.LOCATION_DETAILS\\" translation]",
      }
    }
  />
  <Screen
    component={[Function]}
    name="AddLocation"
    options={
      Object {
        "headerBackTitleVisible": false,
        "headerTitle": "[missing \\"en.LOCATION.ADD_NEW_LOCATION\\" translation]",
        "headerTitleAlign": "left",
        "headerTitleStyle": Object {
          "fontSize": 18,
        },
      }
    }
  />
</StackNavigator>
`;

exports[`Remove aisle render tests renders the bottom sheet remove aisle, non visible, non manager option 1`] = `null`;

exports[`Remove aisle render tests renders the bottom sheet remove aisle, visible, manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="Remove aisle"
/>
`;

exports[`Remove card render tests renders the bottom sheet add card, not visible 1`] = `null`;

exports[`Remove card render tests renders the bottom sheet add card, visible, manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;

exports[`Remove card render tests renders the bottom sheet add card, visible, non manager option 1`] = `
<BottomSheetBaseCard
  image=""
  onPress={[Function]}
  text="hello"
/>
`;
